tools/database/dbc/random_suffix.go

```diff
@@ -21,13 +21,19 @@ func (raw RandomSuffix) ToProto() *proto.ItemRandomSuffix {
 	}
 	for i, effect := range raw.Effects {
 		if effect == 5 || effect == 4 {
-			stat, _ := MapBonusStatIndexToStat(raw.EffectArgs[i])
+			match := false
+			stat, assignedBonusStat := MapBonusStatIndexToStat(raw.EffectArgs[i])
+			match = assignedBonusStat
 			if effect == 4 {
-				stat, _ = MapResistanceToStat(raw.EffectArgs[i])
+				resistanceStat, assignedResistanceStat := MapResistanceToStat(raw.EffectArgs[i])
+				stat = resistanceStat
+				match = assignedResistanceStat
+			}
+			if !match {
+				continue
 			}
 			amount := raw.AllocationPct[i]
 			suffix.Stats[stat] = float64(amount)
-
 		}
```

Comment:
This can be cleaned up a bit:
```
var stat proto.Stat
var matchFound bool

if effect == 5 {
    stat, matchFound = MapBonusStatIndexToStat(raw.EffectArgs[i])
} else if effect == 4 {
    stat, matchFound = MapResistanceToStat(raw.EffectArgs[i])
}

if !matchFound {
    continue
}

amount := raw.AllocationPct[i]
suffix.Stats[stat] = float64(amount)
```